src/pages/iou/IOUModal.js

```diff
@@ -0,0 +1,173 @@
+import React, {Component} from 'react';
+import {View, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import ROUTES from '../../ROUTES';
+import {redirectToLastReport} from '../../libs/actions/App';
+import IOUAmountPage from './steps/IOUAmountPage';
+import IOUParticipantsPage from './steps/IOUParticipantsPage';
+import IOUConfirmPage from './steps/IOUConfirmPage';
+import Header from '../../components/Header';
+import styles from '../../styles/styles';
+import Icon from '../../components/Icon';
+import getPreferredCurrency from '../../libs/actions/IOU';
+import {Close, BackArrow} from '../../components/Icon/Expensicons';
+
+/**
+ * IOU modal for requesting money and splitting bills.
+ */
+const propTypes = {
+
+    /* Onyx Props */
+    // Url currently in view
+    currentURL: PropTypes.string,
+
+    // IOU modal data
+    iouData: PropTypes.shape({
+        isLoadingCurrency: PropTypes.bool.isRequired,
+        isLoadingParticipants: PropTypes.bool.isRequired,
+        createReportInProgress: PropTypes.bool.isRequired,
+    }),
+};
+
+const StepType = {
+    IOUAmount: 'IOUAmount',
+    IOUParticipants: 'IOUParticipants',
+    IOUConfirm: 'IOUConfirm',
+};
+
+const defaultProps = {
+    currentURL: '',
+    iouData: {
+        isLoadingCurrency: true,
+        isLoadingParticipants: true,
+        createReportInProgress: false,
+    },
+};
+
+class IOUModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            steps: [StepType.IOUAmount, StepType.IOUParticipants, StepType.IOUConfirm],
+            currentStepIndex: 0,
+            hasMultipleParticipants: this.props.currentURL === ROUTES.IOU_BILL_SPLIT,
+        };
+
+        this.getTitleForStep = this.getTitleForStep.bind(this);
+        this.navigateToPreviousStep = this.navigateToPreviousStep.bind(this);
+        this.navigateToNextStep = this.navigateToNextStep.bind(this);
+    }
+
+    componentDidMount() {
+        getPreferredCurrency();
+    }
+
+    /**
+     * Returns the title for the currently selected page
+     *
+     * @return {String}
+     */
+    getTitleForStep() {
+        switch (this.state.steps[this.state.currentStepIndex]) {
+            case StepType.IOUAmount:
+                return 'Amount';
+            case StepType.IOUParticipants:
+                return 'Participants';
+            case StepType.IOUConfirm:
+                return 'Confirm';
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Navigate to the next IOU step
+     */
+    navigateToPreviousStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex - 1}));
+    }
+
+    /**
+     * Navigate to the previous IOU step
+     */
+    navigateToNextStep() {
+        this.setState(prevState => ({currentStepIndex: prevState.currentStepIndex + 1}));
+    }
```

Comment:
Don't we need to check there indeed is a next/previous step?

Reply:
Oh, I think we are doing that below?

Reply:
hmmm, not quite for `navigateToNextStep`, right?

Reply:
If the header/buttons are configured correctly it won't not be possible to break, but yeah it's probably a good idea to protect anyway 👍